React/NextJS routing.md

```diff
@@ -0,0 +1,455 @@
+			{children}
+		</a>
+	);
+});
+```
+
+Prefer to use the NextJS `Link` component outside the implementation:
+```tsx
+// BAD
+const Card = (...) => {
+    return (
+        <Link href={href}>{...}</Link>
+    );
+}
+
+<Card href="/company/123" {...} />
+
+
+// GOOD
+const Card = (...) => {
+    return (
+        ...
+    );
+}
+
+<Link href="/company/123">
+    <a>
+        <Card>{...}</Card>
+    </a>
+</Link>
+```
+
+By default, the links created with the `Link` component will prefetch the pages. `Link` uses 
+[`Intersection Observer API`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API)
+to detect when to load with a `rootMargin` of `200px`. At the moment observer values can not be changed (checked in NextJS 12.1.4).
+You can turn off this feature if needed by setting `false` for the prefetch` prop.
+
+For the API reference, please check the official docs for the [`Link` component](https://nextjs.org/docs/api-reference/next/link).
+
+
+## Base path
+
+You can change the base path if your application needs to be under a sub-path of a domain.
+To do that, change the `basePath` property in the `next.config.js` file.
+
+
+## 404 pages
+
+Routing to non-existing pages is working out of the box. 
+Only if there is a `SSR` or a `SSG` then you need to handle it manually. To implement that, you should return: 
+```
+notFound: true
+```
+when you detect a non-existing page, or if for some reason you do not want to show the page content (e.g. the user is
+not authorized).
+It is also important to handle revalidation 
+(for [`SSG`](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration)) otherwise the routes 
+will always show the 404 page.
+
+```tsx
+// SSR pages/article/[articleId].tsx
+
+const ArticlePage = ({ article }) => {
+    return (
+        <>
+            <div>{article.id}</div>
+            <div>{article.title}</div>
+        </>
+    );
+};
+
+export const getServerSideProps = async (context) => {
+    const articleId = context.params.articleId;
+    let article = null;
+
+    try {
+        article = await fetch(`https://my.api.com/article/${articleId}`)
+            .then(response => response.json())
+            .then(data => data);
+    } catch (error) {
+        return {
+            notFound: true,
+        };
+    }
+
+    if (article.id !== articleId) {
+        return {
+            notFound: true,
+        };
+    }
+
+    return {
+        props: {
+            article,
+        },
+    };
+};
+
+export default ArticlePage;
+```
+
+```tsx
+// SSG pages/article/[articleId].tsx
+
+const ArticlePage = ({ article }) => {
+    return (
+        <>
+            <div>{article.id}</div>
+            <div>{article.title}</div>
+        </>
+    );
+};
+
+export const getStaticPaths = async () => {
+    return {
+        paths: [],
+        fallback: 'blocking'
+    }
+};
+
+export const getStaticProps = async (context) => {
+    const articleId = context.params.articleId;
```

Comment:
You are not setting `context.params` in `getStaticPaths`

Reply:
getStaticPaths does not require any paths so you can have empty array. Here the topic is routing, and this part is related to static generation and data fetching mechanisms.